mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -67,12 +67,127 @@ macro_rules! event {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct EndpointConfig {
+    bucket: Option<String>,
+    region: String,
+    use_fips: bool,
+    use_accelerate: bool,
+    use_dual_stack: bool,
+    endpoint: Option<Endpoint>,
+    addressing_style: AddressingStyle,
+}
```

Comment:
I don't think `bucket` nor `region` should live here.

- `bucket` should be passed in per-request.
- `region` should be a member of `S3ClientConfig` I think.

Reply:
`bucket` definitely should be passed in per-request like it is today. For `region`, I think living either here or in `S3ClientConfig` is fine, since regions are pretty related to endpoints.